onnxscript/function_libs/torch_lib/ops/core.py

```diff
@@ -119,20 +120,19 @@ def aten_addcdiv(self: TFloat, tensor1: TFloat, tensor2: TFloat, value: float =
 
 @torch_op("aten::addcmul")
 def aten_addcmul(
-    self: TRealUnlessLowPrecisionFloat,
-    tensor1: TRealUnlessLowPrecisionFloat,
-    tensor2: TRealUnlessLowPrecisionFloat,
+    self: TReal,
+    tensor1: TReal,
+    tensor2: TReal,
     value: float = 1.0,
-) -> TRealUnlessLowPrecisionFloat:
+) -> TReal:
     """addcmul(Tensor self, Tensor tensor1, Tensor tensor2, *, Scalar value=1) -> Tensor
 
     Performs the element-wise multiplication of tensor1 by tensor2, multiplies the
     result by the scalar value and adds it to self.
-
-    f16 and lower will need to be casted to f32 or higher to preserve precision.
     """
 
-    return op.Add(self, op.Mul(op.Mul(tensor1, tensor2), value))
+    # Follow the order in https://github.com/pytorch/pytorch/blob/29e3fddb082b5a14262a7246bc62381a55199d45/aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp#L47
+    return op.Add(self, op.Mul(op.Mul(value, tensor1), tensor2))
```

Comment:
This is less accurate in some test cases (doesn't replicate pytorch result). I guess it depends on the input. What do people think?

Reply:
What do you mean by "less accurate"? For some inputs, the results will be out of the tolerance scope?

Would that be a problem of ort?

Reply:
These are very fundamental ops. I think it deserves an deeper look and maybe file an issue.

Reply:
#811